tests/common/process_utils/mod.rs

```diff
@@ -146,10 +153,13 @@ pub fn run_process_until_response_matches<F: Fn(Output) -> bool>(
     }
 
     if attempts <= 0 {
-        panic!(
-            "{} (tried to connect {} times with {} s interval)",
-            &error_description, &max_attempts, &timeout
-        );
+        return Err(ProcessError::ProcessExited {
+            message: format!(
+                "{} (tried to connect {} times with {} s interval)",
+                &error_description, &max_attempts, &timeout
+            ),
+        });
     }
     println!("Success: {}", &command_description);
```

Comment:
`command_description` is already a reference, as is `error_description` above. Does not matter, it's transparent to the `Display` impl.